Jetcaster/wear/src/main/java/com/example/jetcaster/ui/podcast/PodcastDetailsViewModel.kt

```diff
@@ -62,7 +65,7 @@ class PodcastDetailsViewModel @Inject constructor(
 ) : ViewModel() {
 
     private val podcastUri: String =
-        savedStateHandle.get<String>(PodcastDetails.podcastUri).let {
+        savedStateHandle.get<String>(PodcastDetails.PODCAST_URI).let {
             Uri.decode(it)
```

Comment:
We should check, I suspect decode is done already. But encoding is needed when you use a string template.

Reply:
good question, in https://github.com/android/compose-samples/blob/main/Jetcaster/app/src/main/java/com/example/jetcaster/ui/player/PlayerViewModel.kt#L57 the URI is decoded. @arriolac shall we remove it everywhere?

Reply:
We encode the URI when we navigate (e.g. https://github.com/android/compose-samples/blob/main/Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterAppState.kt#L79) so decoding should be done in the receiver (view model)

Reply:
Isn't that because you are providing a string template, so must be already encoded?

```
    object Player : Screen("player/{$ARG_EPISODE_URI}") {
        fun createRoute(episodeUri: String) = "player/$episodeUri"
    }
```

But I thought the navigation library decodes for you.  Is it different for SavedStateHandle?

https://stackoverflow.com/a/68951045
https://android-review.git.corp.google.com/c/platform/frameworks/support/+/1949864

That maybe explains a bug in one of our screen titles, but a nasty API design if so.

Reply:
Oh TIL that it is already decoded. If so, then yes we wouldn't need to decode. Something to double check @kul3r4